test/meson.build

```diff
@@ -2,13 +2,27 @@ gtest_dep = dependency('gtest', required: false)
 
 nntrainer_test_inc = include_directories('./include')
 
+nntrainer_test_deps = [
+  nntrainer_dep,
+  gtest_dep
+]
+
+# build test util when gtest is found
 if gtest_dep.found()
-  nntrainer_unittest_deps = [
-    nntrainer_dep,
-    gtest_dep
-  ]
+  nntrainer_testutil_lib = static_library(
+    'nntrainer_test_util', 
```

Comment:
no need to be static.

Reply:
Thanks. fixed!

Reply:
There is a segfault on tizen test when I change this to shared_library. I will remove  `do not merge` after investigate the problem!

Reply:
Fixed :)

Reply:
Problem occurs even when I delete the content of `replaceString(...)` 
Meanwhile, `valgrind` is pointing out that invalid free is happening in the arguments `string &from, string &to, string &n`

I surveyed for hours and found that gtest as shared library can be a problem as gtest is recommened to be used as a static library.
There are some options to use gtest as shared library but not working for all the platform I've tested.

Since we have to support multiple platform, I think it'd better leave static for the time being.

For the reference.
https://stackoverflow.com/questions/41404836/gtest-and-gmock-double-free-or-corruption
https://github.com/google/googletest/issues/930

Reply:
Sync.